src/hotspot/share/oops/access.hpp

```diff
@@ -332,19 +332,25 @@ class ArrayAccess: public HeapAccess<IS_ARRAY | decorators> {
                        length);
   }
 
-  static inline void oop_arraycopy(arrayOop src_obj, size_t src_offset_in_bytes,
-                                   arrayOop dst_obj, size_t dst_offset_in_bytes,
-                                   size_t length) {
-    AccessT::oop_arraycopy(src_obj, src_offset_in_bytes, static_cast<const HeapWord*>(nullptr),
-                           dst_obj, dst_offset_in_bytes, static_cast<HeapWord*>(nullptr),
-                           length);
+  [[nodiscard]]
```

Comment:
Is it worth leaving a comment on why the `[[nodiscard]]` annotation was used here?

Reply:
I started to write a comment but they end up being something like:
```
// Make sure that the caller does not accidentally ignore the return value
```

which is sort-of what [[nodiscard]] means, so I'm not sure this is a good comment.

Maybe something talking about why it is important to not forget to check the return value:
```
The caller is responsible to throw an exception on failure
```

WDYT?

Reply:
I added the last sentence but I'm a little bit uncertain about the grammar. Feel free to nit-pick if it sounds weird.

Reply:
In C++20 we can have the message be part of the attribute. `[[nodiscard("The caller is responsible to throw an exception on failure")]]`

Reply:
But the way it says `// The caller is responsible to thow an exception on failure` now. There is an typo.

Maybe `It is the caller's responsibility to throw an exception on failure` or `The caller  is responsible for throwing an exception on failure`